ethcore/engine/src/engine.rs

```diff
@@ -188,6 +188,14 @@ pub trait Engine: Sync + Send {
 	/// Returns the engine's current sealing state.
 	fn sealing_state(&self) -> SealingState { SealingState::External }
 
+	/// Called in `miner.chain_new_blocks` if the engine wishes to `update_sealing`
+	/// after a block was recently sealed, and there are local pending tx in the pool.
```

Comment:
I think we don't need to guard this by additional check with the transaction pool. Just triggering a reseal will cause the `generate_seal` method to check if there are transactions or not:
https://github.com/paritytech/parity-ethereum/blob/81ca599f2adf7737c4c57f49fce94a304c6fc242/ethcore/engines/instant-seal/src/lib.rs#L78

Reply:
^^ Still worth updating.